### PR TITLE
Use sbt-github-actions

### PR DESCRIPTION
**Pull request by amesgen**
Supersedes and closes #3825

Can't test publishing in advance (as usual).

It might be a good idea to (eventually) put more of this into sbt-http4s-org for code reuse.

### CODE REVIEW

**Comment by amesgen** [comment]
MIMA is disabled as it fails right now, and I don't know why: [logs](https://github.com/amesgen/http4s/runs/1388170126)

Also TODO: adapt mergify

File: project/Http4sPlugin.scala [hunk 1]
```diff
@@ -31,11 +31,11 @@ object Http4sPlugin extends AutoPlugin {
     ThisBuild / http4sApiVersion := (ThisBuild / version).map {
       case VersionNumber(Seq(major, minor, _*), _, _) => (major.toInt, minor.toInt)
     }.value,
-  )
+  ) ++ sbtghactionsSettings
 
   override lazy val projectSettings: Seq[Setting[_]] = Seq(
     scalaVersion := scala_213,
-    crossScalaVersions := Seq(scala_213, scala_212),
+    ThisBuild / crossScalaVersions := Seq(scala_213, scala_212),
```
Comment (rossabaker) [inline project/Http4sPlugin.scala:38]: Can a `ThisBuild` go in `projectSettings`?  I thought that needed to be in `buildSettings`?

File: project/Http4sPlugin.scala [hunk 2]
```diff
@@ -227,6 +227,73 @@ object Http4sPlugin extends AutoPlugin {
...
+    inThisBuild(Http4sOrgPlugin.githubActionsSettings ++ Seq(
+      githubWorkflowBuild := Seq(
+        WorkflowStep
+          .Sbt(List("scalafmtCheckAll"), name = Some("Check formatting")),
+        WorkflowStep.Sbt(List("headerCheck", "test:headerCheck"), name = Some("Check headers")),
+        WorkflowStep.Sbt(List("test:compile"), name = Some("Compile")),
+        // TODO reenable
+        // WorkflowStep.Sbt(List("mimaReportBinaryIssues"), name = Some("Check binary compatibility")),
+        WorkflowStep.Sbt(
+          List("unusedCompileDependenciesTest"),
```
Comment (rossabaker) [inline project/Http4sPlugin.scala:266]: I'd still like to turn on the other test eventually, but that's a separate ticket.

File: project/Http4sPlugin.scala [hunk 3]
```diff
@@ -227,6 +227,73 @@ object Http4sPlugin extends AutoPlugin {
...
+          name = Some("Check explicit dependencies")),
+        WorkflowStep.Sbt(List("test"), name = Some("Run tests")),
+        WorkflowStep.Sbt(List("doc"), name = Some("Build docs")),
+        setupHugoStep,
+        siteBuildStep("website"),
+        siteBuildStep("docs")
```
Comment (rossabaker) [inline project/Http4sPlugin.scala:272]: These used to be their own job, which meant they could run in parallel.  It also meant they could update if one of the safety checks failed, so maybe that was bad.

**Review by rossabaker — COMMENTED** [review]
Nicely done. This is probably going to fail somewhere, but these things always require iteration, and it's a great start.

We should target series/0.21 with this instead.  Most work is still starting there, and I think those builds will start failing on November 16 without this.

File: project/Http4sPlugin.scala (hunk 1 above)
Comment (amesgen) [reply project/Http4sPlugin.scala:38]: I have only a very rudimentary understanding of the differences between `projectSettings` and `buildSettings`. I had to add `ThisBuild / ` for sbt-github-actions to pick it up, and it appears to work(?). I can move it to though, it seems to continue to work locally with it being moved.

**Review by rossabaker — COMMENTED** [review]
I thought MiMa might be related to fetch depth and not finding the tags, but it looks like the depth is 0 on both checkouts. :thinking:

**Comment by amesgen** [comment]
The fail-fast behavior is not optimal, will create a PR to sbt-github-actions to allow to disable it.

I am confused about the last build error:
```
sbt.librarymanagement.ResolveException: Error downloading org.http4s:http4s-core_2.12:1.0.0-M6+9-6595f52a-SNAPSHOT
```

**Comment by amesgen** [comment]
Wait, I should have not used the GitHub UI branch retarget feature... Will close and reopen

**Pull request closed without merging**